Source/Blazorise/Components/ModalProvider/ModalProvider.razor.cs

```diff
@@ -0,0 +1,131 @@
+﻿#region Using directives
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Blazorise.Modules;
+using Blazorise.States;
+using Blazorise.Utilities;
+using Microsoft.AspNetCore.Components;
+using Microsoft.JSInterop;
+#endregion
+
+namespace Blazorise
+{
+    /// <summary>
+    /// A modal provider to be set at the root of your app, providing a programatic way to invoke modals with custom content by using ModalService.
+    /// </summary>
+    public partial class ModalProvider
+    {
+        #region Members
+
+        private Modal modalRef;
+        private RenderFragment childContent;
+        private bool useModalDefinition = true; //TOOD : Choose a better name
```

Comment:
This will probably be set through the options, right?

Reply:
Ye. Global options and then overriden options when calling `Show`